### ISSUE TITLE
Feature Request: Autocomplete for tests / scripts

### ISSUE THREAD

**Issue by vdespa** [a85/POSTMan-Chrome-Extension#3919]
It would be nice to have an autocomplete functionality for the scripts. Especially the new pm API and the chai assertions are a bit harder to master and having autocomplete would make our lives easier.

I would just like to know if this is something you are planning for.

Thanks for your work.

**Comment by thecuriousdev** [comment]
@vdespa thanks for this feature request, we'll keep you posted here.

**Comment by rysbilinski** [comment]
+1

**Comment by mccannt** [comment]
This is something that I feel would be quite useful to have for the same reasons that @vdespa has mentioned. 
Especially for dev/qe folks who are new to the tool and used to IDE's that allow for things like this. 

I believe this can help drive use and adoption for some as it can take the guess work out of things for in-built libs.

**Comment by adsm1987** [comment]
I totally agree, I'm used to IDE and I miss this feature.


Thanks

**Comment by buehler** [comment]
Since I'm missing this feature as well, I've got an idea:
When working on another project, I needed an "online" editor as well. Since postman is an electron app, it could be an option to customize and include the monaco editor which powers vscode right now.

It's easy customizable and has features as intellisense and other elements out of the box.

https://microsoft.github.io/monaco-editor/

**Comment by svcat** [comment]
Yep, as a qa starting to do automation, I see this feature very useful.

**Comment by rk-7** [comment]
This will be awesome! I'm waiting for this feature since it helps you write code fast and without silly mistakes.

**Comment by mauricioleoncio** [comment]
Hi there! any update on this feature? :)

**Comment by niikoo** [comment]
Is there an ETA? @thecuriousdev

**Comment by sajalshres** [comment]
Any update? Its been open for a while.

**Comment by vvs11** [comment]
This is definitely in our roadmap and currently, we are figuring out the technical constraints around building it in the App. Will keep you notified on further developments on this

**Comment by pankajk07** [comment]
Hi all, we have added this feature and is available in our [Canary](https://www.postman.com/downloads/canary/) app. Do give it a try and please provide feedback.

P.S. The stable app and canary app can run in parallel

**Comment by vdespa** [comment]
Great to finally see this. I have used it a bit and here are some ideas based on my tests done using Version 7.28.0-canary03 (7.28.0-canary03):

1. Make the detailed method/property documentation the default view (if not already) as it helps understand what each method is (so the users don't have to click on the small (i) icon.

![image](https://user-images.githubusercontent.com/1197635/86866892-ac794200-c086-11ea-9976-26c5a57b5e7f.png)

2. pm.expect(...)to. should have more options here, like to.eql

![image](https://user-images.githubusercontent.com/1197635/86866974-d2064b80-c086-11ea-8c00-a3397af4243b.png)

3. Is there a way to trigger autocomplete to show everything that is in the current scope without typing something?

4. It should be super helpful to have an autocomplete on the pm.response.json() object with the last retrieved response. This may help many many people understand how to navigate complex responses. 

5. I don't see any autocomplete for the build-in array methods (concat, reduce, find, ...).

![image](https://user-images.githubusercontent.com/1197635/86868060-d6cbff00-c088-11ea-95cc-451ac82bb9e9.png)

6. It seems that the autocomplete does not recognize the Standard built-in objects (Object, Array, ...), Function properties
 (eval, parseInt) and maybe others as well?

7. 3rd party libraries like CryptoJS or cheerio are not detected. 

8. No autocomplete with anything pulled in with require

![image](https://user-images.githubusercontent.com/1197635/86868923-7938b200-c08a-11ea-9203-cd388f7a0a4e.png)

**Comment by pankajk07** [comment]
Hello everyone. Autocompletion in script editors is now available in the stable channel on the latest release (`7.29`).

@vdespa, thanks for taking the time to try out the `Canary` build and providing us with detailed feedback. We have made the following improvements which should be available on both `Canary` and `Stable` channel.

1. You can trigger autocomplete at any location by pressing `Ctrl + Space`
2. Array methods should show up in autocomplete now
3. Standard built-in objects like Object, Arrays, Strings should now be recognised and their methods should show up in autocomplete.
4. Global methods like eval, parseInt , setTimeout should show up in autocomplete now
5. We have included support NodeJS modules like path, assert. We will add support for thirdparty modules as well.

We will continue to make more improvements to this feature and address the rest of the feedback as well.

**Comment by vdespa** [comment]
Great job, hope to see other improvements. :)
